sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala

```diff
@@ -747,7 +747,7 @@ class SQLContext private[sql](
    * only during the lifetime of this instance of SQLContext.
    */
   private[sql] def registerDataFrameAsTable(df: DataFrame, tableName: String): Unit = {
-    catalog.registerTable(TableIdentifier(tableName), df.logicalPlan)
+    catalog.registerTable(SqlParser.parseTableIdentifier(tableName), df.logicalPlan)
```

Comment:
We have removed ````SqlParser``` recently. Could you merge/rebase to the most recent master, and use the ```sqlParser``` variable for this?